oaipmh/src/main/java/whelk/export/servlet/Helpers.java

```diff
@@ -151,6 +151,18 @@ else if (updatedCollection.equals("hold"))
 
         }
 
+        private boolean hasCardChanged(Document updated) {
+            Document previousVersion = OaiPmh.s_whelk.getStorage().load(updated.getShortId(), "-1");
+            if (previousVersion == null) {
+                return true;
+            }
+            
+            var jsonLd = OaiPmh.s_whelk.getJsonld();
+            var oldCard = jsonLd.toCard(previousVersion.data);
+            var newCard = jsonLd.toCard(updated.data);
+            return !oldCard.equals(newCard);
```

Comment:
This _ought_ to be stable enough (and is certainly simple enough). It hinges on whether the `equals` of the List and Map implementations are "deep equals", which is should be the case for [AbstractList](https://docs.oracle.com/javase/1.5.0/docs/api/java/util/AbstractList.html#equals(java.lang.Object)) and Map (entry sets) see [AbstractSet](https://docs.oracle.com/javase/1.5.0/docs/api/java/util/AbstractSet.html#equals(java.lang.Object)) (since at least 1.5.0). It effectively depends on implementation though; so even if wee're good now, _if_  the JSON parser would change, we would need some test cases to ensure this contract holds?

Reply:
It's part of the contract of the List and Map interfaces. So unless we change to something that maps JSON arrays and objects to something different I thing we are fine.
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Map.html#equals(java.lang.Object)
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/List.html#equals(java.lang.Object)

Reply:
Switching to our `DocumentComparator.isEqual` would treat arrays/lists as unordered where they are.